clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/view/GoogleClusterProvider.groovy

```diff
@@ -77,26 +78,37 @@ class GoogleClusterProvider implements ClusterProvider<GoogleCluster.View> {
   }
 
   Map<String, Set<GoogleCluster.View>> getClusters(String applicationName, boolean includeInstanceDetails) {
-    GoogleApplication.View application = applicationProvider.getApplication(applicationName)
+    CacheData applicationCacheData = applicationProvider.getApplicationCacheData(applicationName)
 
-    def clusterKeys = []
-    application?.clusterNames?.each { String accountName, Set<String> clusterNames ->
-      clusterNames.each { String clusterName ->
-        clusterKeys << Keys.getClusterKey(accountName, applicationName, clusterName)
-      }
-    }
-
-    // TODO(jacobkiefer): Avoid parsing instance keys into map just to re-serialize?
-    Set<String> allApplicationInstanceKeys = includeInstanceDetails ? application?.instances?.collect { Keys.getInstanceKey(it.account, it.region, it.name) } : [] as Set
-
-    List<GoogleCluster.View> clusters = cacheView.getAll(
-        CLUSTERS.ns,
-        clusterKeys,
-        RelationshipCacheFilter.include(SERVER_GROUPS.ns)).collect { CacheData cacheData ->
-      clusterFromCacheData(cacheData, allApplicationInstanceKeys)
+    Set<String> applicationClusterIdentifiers = applicationProvider.getApplicationClusterIdentifiers(applicationCacheData)
+    Collection<CacheData> clusterCacheData = cacheView.getAll(
+      CLUSTERS.ns,
+      applicationClusterIdentifiers,
+      RelationshipCacheFilter.include(SERVER_GROUPS.ns)
+    )
+
+    Set<String> applicationInstanceIdentifiers = includeInstanceDetails ?
+      applicationProvider.getApplicationInstanceIdentifiers(applicationCacheData) :
+      Collections.<String>emptySet()
+    Collection<CacheData> applicationInstanceCacheData = instanceProvider.getInstanceCacheData(applicationInstanceIdentifiers)
```

Comment:
supernit: I kept getting `applicationCacheData` and `applicationInstanceCacheData` mixed up. IMO `instanceCacheData` is just as clear but more distinct. Feel free to ignore if you disagree :grinning:

Reply:
:+1: :+1: updated!